conda/deprecations.py

```diff
@@ -285,6 +335,20 @@ def _get_module(self, stack: int) -> tuple[ModuleType, str]:
 
         try:
             frame = sys._getframe(2 + stack)
+            # Shortcut finding the module by manually inspecting loaded modules.
+            if hasattr(frame, "f_code") and hasattr(frame.f_code, "co_filename"):
```

Comment:
duck typing instead?

and looking at this again it appears that 338-354 should actually be defined outside this try-except:

```python
try:
    frame = sys._getframe(2 + stack)
except IndexError:
    # IndexError: 2 + stack is out of range
    raise DeprecatedError("unable to determine the calling module")

try:
    filename = frame.f_code.co_filename
except AttributeError:
    # AttributeError: frame.f_code.co_filename is undefined
    pass
else:
    for module in sys.modules.values():
        ...

import inspect
...
```

Reply:
> duck typing instead?

I don't mind either way; added your suggestion!

> and looking at this again it appears that 338-354 should actually be defined outside this try-except:

Ah, I didn't even look at the exception type. Makes sense -- added!